contracts/cryptoblades.sol

```diff
@@ -225,6 +228,8 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
     mapping(address => mapping(uint256 => uint256)) public userVars;
 
     SpecialWeaponsManager public specialWeaponsManager;
+    mapping(uint256 => address) public links;
+    mapping(address => uint256) public claimWeaponData;
```

Comment:
I'd prefer not to introduce the claimWeaponData variable.
Technically we can call safeRandoms.hasSingleSeedRequested(msg.sender, seed) to check for that.
There's an external function called "encode" on safeRandoms that takes an array to assemble the seed from different pieces, this way we don't have to make the getSeed functions public in cryptoblades/blacksmith.

Let me know if that works

Reply:
claimWeaponData wasn't introduced to replace hasSeed or something like that, it was introduced to have knowledge about what weapon (and how many weapons) to mint during mintWeapon call. Thanks to this, we can know in frontend that player has "any" seed and use that knowledge to highlight claim button. Without it we would have to check 10x for the seeds and decide which one is present.

Reply:
Fair point. I'd probably be crazy enough to go for a halfway measure where we called 5 times on saferandoms with a 1-10 iteration for amount in the function each but i'm not gonna send us down that rabbit hole.

But actually I just realised we have a thing called userVars. How about adding a new key constant for claimWeaponData, and using that instead of adding a new mapping?